server/database/db_connection.js

```diff
@@ -0,0 +1,25 @@
+const mongoose = require('mongoose');
+
+// load .env only on development
+if (!process.env.TRAVIS && !process.env.ENV !== 'production') {
+  require('dotenv').config()
+}
+
+const dbConnection = () => {
+  let { MONGOURI: mongoURI } = process.env;
+
+  if (process.env.NODE_ENV === 'test') {
+    // change MONGOURI to testDBurl
+    mongoURI = process.env.MONGOURI_TEST; 
+  }
+  // create DB connection
+  return mongoose.connect(
+    mongoURI,
+    { 
+      useNewUrlParser: true,
+      useCreateIndex: true,
+    }
+  );
```

Comment:
```suggestion
  )
  .catch(err => debug('meaningful error message'), err.stack);
```
I realize I probably butchered the way we should use debug. @RamyAlshurafa now is a good time to show me how debug should be used.